post-publish-processor/src/main/scala/org/sunbird/job/postpublish/helpers/BatchCreation.scala

```diff
@@ -144,12 +144,27 @@ trait BatchCreation {
     }
   }
 
+  def isActivityBatchExists(metadata: java.util.Map[String, AnyRef]): Boolean = {
+    val batchesStr = metadata.get("batches").asInstanceOf[String]
+    if (StringUtils.isBlank(batchesStr)) return false
+    try {
+      val batches = JSONUtil.deserialize[java.util.List[java.util.Map[String, AnyRef]]](batchesStr)
+      val batchExists = CollectionUtils.isNotEmpty(batches)
+      if(batchExists) logger.info("Activity has existing batches. Skipping creation.")
+      batchExists
+    } catch {
+      case e: Exception =>
+        logger.error("Failed to parse 'batches' metadata field.", e)
+        false
```

Comment:
```suggestion
        true
```

Send "true" so that, it won't create the batch when there is some exception.